docs/reference/graphql.md

```diff
@@ -0,0 +1,170 @@
+
+```
+query getUser($id: ID!) { 
+    user(id: $id) {
+        name 
+        email 
+    }
+}
+```
+
+#### `graphql_variables`
+
+A json of key value pairs of variables used in the GraphQL query. The connector will substitute the variables in the query with the values provided here and make a GraphQL query to the source.
+
+Example: If the GraphQL query is `query getUser($id: ID!) { user(id: $id) { name } }` and the value of `graphql_variables` is `{"id": "123"}`, the final query will be `query getUser { user(id: "123") { name } }` and the connector will make a GraphQL query to the source with the modified query.
+
+#### `graphql_object_list` (required)
+
+A JSON mapping between GraphQL response objects to index and their ID fields. The connector will fetch data for each object (JSON key) and use the provided ID field (JSON value) to index the object into Elasticsearch. The connector will index all fields for each object specified in the mapping. Use a dot `(.)` notation to specify the full path from the root of the GraphQL response to the desired object.
+
+Example: 
+
+```
+If the GraphQL query is `query getUser { organization { users{ user_id name email} } }` which is fetching all available users from the source and we want to index every user as separate document in that case configure this field as below.
+{
+    "organization.users": "user_id"
+}
+
+Here user_id is unique in all the user document so configure user_id as a value to `organization.users`.
+```
+
+Note: The path provided in this field should only contain Json Objects and not lists.
+
+#### `headers`
+
+Json of custom headers to be sent with each GraphQL request. Example:
+
+```
+{
+    "content-type": "Application/json"
+}
+```
+
+#### `pagination_model` (required)
+
+This field specifies the pagination model to be used by the connector. The connector supports `No pagination` and `Cursor-based pagination` pagination models. For cursor-based pagination, add `pageInfo {endCursor hasNextPage}` and an `after` argument variable in your query at the desired node (`Pagination key`). Use `after` query argument with a variable to iterate through pages. Detailed examples and setup instructions are available in the docs. The default value for this field is `No pagination`. Example:
+
+```
+For `Cursor-based pagination`, the query should look like below example:
+
+query getUsers($cursor: String!) {
+    sampleData { 
+        users(after: $cursor) {
+            pageInfo {
+                endCursor
+                hasNextPage
+            }
+            nodes {
+                first_name
+                last_name
+                address
+            } 
+        } 
+    } 
+}
+and the value of `pagination_key` is `sampleData.users` so, it must contain `pageInfo {endCursor hasNextPage}` and `after` argument with a variable in case of `Cursor-based pagination`.
```

Comment:
this should be outside of the code block

Reply:
the whole section is inside a code block

Reply:
Oh yes, thank you for pointing that out.